astroquery/desi/core.py

```diff
@@ -0,0 +1,140 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+
+import os
+import shutil
+import tempfile
+import urllib.error
+
+from astropy.io.fits import HDUList
+import requests
+from astropy.table import Table, vstack
+
+from astroquery.exceptions import NoResultsWarning
+from ..query import BaseQuery
+from ..utils import commons
+from ..utils import async_to_sync
+from . import conf
+
+import pyvo as vo
+import numpy as np
+
+__all__ = ['DESILegacySurvey', 'DESILegacySurveyClass']
+
+
+@async_to_sync
+class DESILegacySurveyClass(BaseQuery):
+    URL = conf.server
+    TIMEOUT = conf.timeout
+
+    def query_region_async(self, coordinates, radius, data_release=9):
+        """
+        Queries a region around the specified coordinates.
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates`.
+            coordinates around which to query
+        radius : str or `astropy.units.Quantity`.
+            the radius of the cone search
+
+        Returns
+        -------
+        response : `requests.Response`
+            The HTTP response returned from the service.
+            All async methods should return the raw HTTP response.
+        """
+
+        url = 'https://datalab.noirlab.edu/tap'
+        tap_service = vo.dal.TAPService(url)
+        qstr = "SELECT all * FROM ls_dr" + str(data_release) + ".tractor WHERE dec>" + str(coordinates.dec.deg - radius.deg) + " and dec<" + str(
+            coordinates.dec.deg + radius.deg) + " and ra>" + str(coordinates.ra.deg - radius.deg / np.cos(coordinates.dec.deg * np.pi / 180.)) + " and ra<" + str(
+            coordinates.ra.deg + radius.deg / np.cos(coordinates.dec.deg * np.pi / 180))
+
+        tap_result = tap_service.run_sync(qstr)
+        tap_result = tap_result.to_table()
+        # filter out duplicated lines from the table
+        mask = tap_result['type'] != 'D'
+        filtered_table = tap_result[mask]
+
+        return filtered_table
+
+    def get_images_async(self, position, survey, coordinates=None, data_release=9,
+                         projection=None, pixels=None, scaling=None,
+                         sampler=None, resolver=None, deedger=None, lut=None,
+                         grid=None, gridlabels=None, radius=None, height=None,
+                         width=None, cache=True, show_progress=True, image_band='g'):
+        """
+        Returns
+        -------
+        A list of context-managers that yield readable file-like objects
+        """
+
+        image_size_arcsec = radius.arcsec
+        pixsize = 2 * image_size_arcsec / pixels
+
+        image_url = 'https://www.legacysurvey.org/viewer/fits-cutout?ra=' + str(position.ra.deg) + '&dec=' + str(position.dec.deg) + '&size=' + str(
+            pixels) + '&layer=ls-dr' + str(data_release) + '&pixscale=' + str(pixsize) + '&bands=' + image_band
+
+        print("image_url: ", image_url)
```

Comment:
Suggest offering a way to suppress this print statement by having a verbose argument or similar.